### PR TITLE
feat: X8 pane supervision — daemon-side init system for agent panes

### PR DESCRIPTION
**Pull request by openwong2kim**
## X8 pane supervision — the daemon-side init system for agent panes

Adds supervision to `wmux.json` pane specs: an exec-style agent loop that exits is automatically relaunched by the daemon. The principle is **"a house for the loop, not the loop"** — the daemon owns only *when* to relaunch and when to give up; goals and stopping conditions stay with the agent CLI inside the pane (substrate neutrality). Reuses daemon respawn (#54), the X5 trust gate, and the X4 CLI, so it adds the init primitive without new infrastructure.

### What's in it
- **daemon core** (`PaneSupervisor`, `execWrapper`): exec-style panes run the command AS the pane root (`pwsh -Command "<cmd>; <exit-tail>"`, `cmd /d /s /c`, `sh -lc`), so `session:died` carries the command's own exit code and a recovery/restart replay relaunches the command itself — not an empty shell. `restart: on-failure | always` with exponential backoff (1s → 30s) and a runaway guard (StartLimitBurst-style: N consecutive short-lived runs → supervision stops, sticky + persisted; only a manual rearm resumes).
- **main / IPC + schema**: `wmux.json` supervision schema, `session.restarted` / `supervision.changed` broadcasts, a `PTY_RESTARTED` relay that re-attaches the renderer without running the died-path teardown, and `supervise.stop` / `supervise.rearm` RPCs (renderer-only, capability-gated — external pipe clients are blocked).
- **surfaces**: sidebar ⟳ / ⟳! badge (uptime / restarts), `wmux list --json` supervision field, trust-dialog "auto-restarts" disclosure, pane-menu Stop / Rearm, and an agent.lifecycle event tee.

### Trust model (what makes it safe)
- `session:died` (PTY exited on its own) → policy evaluation.
- `session:destroyed` (user closed the pane) → disarm. A supervisor that only ever restarts on `died` is structurally unable to resurrect a pane the user closed.
- Supervisor-initiated restarts go through silent tombstone removal (never `destroySession`), so a restart can't masquerade as a user close. User intent is expressed by closing the pane or the explicit stop action — never inferred from an exit code (Ctrl+C is 130 on POSIX but 0xC000013A on Windows, so code-based detection is platform-fragile).

### Verification
- **Unit**: `PaneSupervisor` / `execWrapper` / `wmuxProjectConfig` / `supervisionSlice`, plus full suite green (3176/3176).
- **Live dogfood 27/27** (dev build over CDP + suffix-scoped daemon RPC as ground truth): restart:always revival, on-failure classification (exit 0 → no restart / exit 1 → restart), pane-close = no resurrection (the trust test), close-during-backoff, runaway guard + rearm, **daemon-restart survival** (kill the daemon → respawn replays the exec session → supervision re-arms, the loop grows with a new pid — the launch demo anchor), surfaces, and Ctrl+C semantics.

## Summary by CodeRabbit

* **New Features**
  * Supervised panes: declare exec-style panes in wmux.json to auto-restart processes (on-failure | always) with configurable limits; supervision persists across daemon/app restarts and excludes plain panes.
  * Renderer controls: supervision badge, context-menu Stop/Rearm, terminal restart markers, and project-config supervision badges; rearm/stop controls available from UI.

* **Documentation**
  * API reference and event docs updated with new daemon RPCs/events and supervision metadata.

* **Tests**
  * Added test coverage for supervision, exec-wrapper behavior, and lifecycle edge cases.

### CODE REVIEW

File: docs/api/reference.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ returns `EPERM`. Wire framing: newline-delimited JSON, one object per line.
 
 ## RPC methods
 
-Total: **97** methods (`ALL_RPC_METHODS` in
+Total: **99** methods (`ALL_RPC_METHODS` in
 `src/shared/rpc.ts`). Capability and risk class are read from
 `src/main/mcp/methodCapabilityMap.ts`:
```
[the remark this answers is not in the document]
Comment (openwong2kim) [reply docs/api/reference.md:31]: pass
